static/app/views/performance/database/useSelectedDurationAggregate.tsx

```diff
@@ -34,12 +33,16 @@ export function useSelectedDurationAggregate(): Result {
 
   const location = useLocation<Query>();
 
-  let selectedAggregate = decodeScalar(
+  let selectedAggregate: Aggregate;
+  const aggregateFromURL = decodeScalar(
     location.query.aggregate,
     previouslySelectedAggregate
   );
 
-  if (!availableAggregates.includes(selectedAggregate)) {
+  // `availableAggregates` is typed `as const` so I have to cast to unknown to allow comparison to string
+  if ((availableAggregates as unknown as string[]).includes(aggregateFromURL)) {
+    selectedAggregate = aggregateFromURL as Aggregate;
```

Comment:
m: It was a bit hard for me to understand what the `includes` call was doing at first. What do you think of something like this?

```ts
const aggregateFromURL = decodeScalar(
  location.query.aggregate,
  previouslySelectedAggregate
);

const selectedAggregate = isAggregate(availableAggregates, aggregateFromURL)
  ? aggregateFromURL
  : DEFAULT_DURATION_AGGREGATE;

return [selectedAggregate, setSelectedAggregate];
```

where `isAggregate` is a helper like:

```ts
function isAggregate(
  availableAggregates: Aggregate[],
  maybeAggregate: string
): maybeAggregate is Aggregate {
  // couldn't think of a way around the cast here unfortunately, but still should be type safe!
  return availableAggregates.includes(maybeAggregate as Aggregate);
}
```

Reply:
Works for me! I had a version like that at some point but it got lost in between refactors. I'll bring it back 👍🏻